src/cctz.h

```diff
@@ -249,8 +278,77 @@ std::string Format(const std::string& format, const time_point& tp,
 //   "00.x" -> 00.x  // exact
 //
 // Errors are indicated by returning false.
+//
+// Example:
+//   const cctz::TimeZone tz = ...
+//   std::chrono::system_clock::time_point tp;
+//   if (cctz::Parse("%Y-%m-%d", "2015-10-09", tz, &tp)) {
+//     ...
+//   }
+template <typename D>
 bool Parse(const std::string& format, const std::string& input,
-           const TimeZone& tz, time_point* tpp);
+           const TimeZone& tz, time_point<D>* tpp);
+
+}  // namespace cctz
+
+//
+// IMPLEMENTATION DETAILS
+//
+namespace cctz {
+
+namespace internal {
+// Floors tp to a second boundary. Optionally returns subseconds.
+template <typename D, typename Subseconds = std::chrono::nanoseconds>
+inline time_point<seconds64> FloorSeconds(const time_point<D>& tp,
+                                          Subseconds* subseconds = nullptr) {
```

Comment:
The only difference would be that the body of `BreakTime()` below would become:

```diff
-  return BreakTime(internal::FloorSeconds(tp), tz);
+  return BreakTime(internal::FloorSeconds(tp, static_cast<D*>(nullptr)), tz);
```
Since defaulting the second parameter doesn't cost anything, it seemed reasonable. It doesn't matter either way to me. I'll change it if you prefer.